tiledb/sm/cpp_api/type.h

```diff
@@ -93,6 +93,13 @@ struct type_to_tiledb<std::byte> {
   static constexpr const char* name = "BLOB";
 };
 
+template <>
+struct type_to_tiledb<bool> {
+  using type = bool;
+  static const tiledb_datatype_t tiledb_type = TILEDB_BOOL;
+  static constexpr const char* name = "BOOL";
+};
```

Comment:
This fixes the immediate problem, but not the root cause. It's a design error that `template <typename T>
struct type_to_tiledb`, the original declaration, is able to instantiate at all. That class should have a single `static_assert(false, ...)` statement in it. The removes the possibility of default behavior. String types should then have their own specializations like every other type does.

In other words, the design error is that it's an "opt-out" system, that is, there's a default that needs to be opted out of. All things involved the `Datatype` enumeration should be "opt-in", with no behavior specified with an explicit type dependency.

Reply:
Just FYI that this change _is_ on the critical path for making the (initial) tiledbsoma R 1.0 release.  So it would be nice if we could possibly merge this _and_ flag a story for a future refactor.

Reply:
@eddelbuettel @eric-hughes-tiledb I'll file a tracking story

Reply:
sc-28086

Reply:
https://app.shortcut.com/tiledb-inc/stories/space/21943

Reply:
@ihnorton lol! :)